src/patterns/ec2-app.ts

```diff
@@ -94,51 +125,108 @@ function restrictedCidrRanges(ranges: IPeer[]) {
+ *
+ * Example usage for a public facing app (open to the internet):
  * ```typescript
  * new GuEc2App(stack, {
- *       applicationPort: GuApplicationPorts.Node,
- *       app: "app-name",
- *       // ONLY allows access to CIDR ranges provided
- *       access: { scope: AccessScope.RESTRICTED, cidrRanges: [Peer.ipv4("192.168.1.1/32"), Peer.ipv4("8.8.8.8/32")] },
- *       certificateProps:{
- *           [Stage.CODE]: {
- *             domainName: "code-guardian.com",
- *             hostedZoneId: "id123",
- *           },
- *           [Stage.PROD]: {
- *             domainName: "prod-guardian.com",
- *             hostedZoneId: "id124",
- *           },
- *       },
- *       monitoringConfiguration: { noMonitoring: true },
- *       userData: "",
+ *   applicationPort: 1234,
+ *   app: "app-name",
+ *   access: { scope: AccessScope.PUBLIC },
+ *   certificateProps:{
+ *     [Stage.CODE]: {
+ *       domainName: "code-guardian.com",
+ *       hostedZoneId: "id123",
+ *     },
+ *     [Stage.PROD]: {
+ *       domainName: "prod-guardian.com",
+ *       hostedZoneId: "id124",
+ *     },
+ *   },
+ *   monitoringConfiguration: {
+ *     tolerated5xxPercentage: 1,
+ *     snsTopicName: "alerts-topic-for-my-team",
+ *   },
+ *   userData: {
+ *     distributable: {
+ *       fileName: "app-name.deb",
+ *       executionStatement: `dpkg -i /app-name/app-name.deb`,
+ *     }
+ *   },
  * });
+ * ```
  *
- * // For public-facing applications
+ * Example usage for an app which is locked down to specific IP ranges:
+ * ```typescript
  * new GuEc2App(stack, {
- *       applicationPort: GuApplicationPorts.Node,
- *       app: "app-name",
- *       // ONLY allows access to CIDR ranges provided
- *       access: { scope: AccessScope.PUBLIC },
- *       certificateProps:{
- *           [Stage.CODE]: {
- *             domainName: "code-guardian.com",
- *             hostedZoneId: "id123",
- *           },
- *           [Stage.PROD]: {
- *             domainName: "prod-guardian.com",
- *             hostedZoneId: "id124",
- *           },
- *       },
- *       monitoringConfiguration: { noMonitoring: true },
- *       userData: "",
+ *   applicationPort: 1234,
+ *   app: "app-name",
+ *   access: {
+ *     scope: AccessScope.RESTRICTED,
+ *     cidrRanges: [Peer.ipv4("192.168.1.1/32"), Peer.ipv4("8.8.8.8/32")],
+ *   },
+ *   certificateProps:{
+ *     [Stage.CODE]: {
+ *       domainName: "code-guardian.com",
+ *       hostedZoneId: "id123",
+ *     },
+ *     [Stage.PROD]: {
+ *       domainName: "prod-guardian.com",
+ *       hostedZoneId: "id124",
+ *     },
+ *   },
+ *   monitoringConfiguration: {
+ *     tolerated5xxPercentage: 1,
+ *     snsTopicName: "alerts-topic-for-my-team",
+ *   },
+ *   userData: {
+ *     distributable: {
+ *       fileName: "app-name.deb",
+ *       executionStatement: `dpkg -i /app-name/app-name.deb`,
+ *     }
+ *   },
  * });
  * ```
- * */
+ *
+ * ### Customising the infrastructure generated by this pattern
```

Comment:
minor: I wonder if the escape hatch information should also live as an individual markdown file in the repo that we can link to in chat etc in future?

Reply:
I was wondering about this too. I guess we will need to add similar escape hatch behaviour for the lambda-based patterns in the future. Perhaps we can generalise the style and docs at that point?